File: Linear_Regression/Simple_linear_regression/Simple_linear_regression.py
# -*- coding: utf-8 -*-
"""
Created on Sun Nov  1 18:42:04 2020

@author: Acer
"""


import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#creating working directory and read data

dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[: ,:-1].values
y=dataset.iloc[: , 1].values

#spliting of data 


from sklearn.model_selection import train_test_split 
X_train ,X_test ,y_train ,y_test = train_test_split(X,y, test_size=1/3 ,random_state = 0)

#fiting linear simple regression to te training data
from sklearn.linear_model import LinearRegression
regressor=LinearRegression()
regressor.fit(X_train, y_train)

#predicting test result
y_pred = regressor.predict(X_test)

#Visualization of training set 

plt.scatter(X_train, y_train, color='red')
plt.plot(X_train , regressor.predict(X_train),color='blue')
plt.title('Salary vs Experession(Training set)')
plt.xlabel('Experience')
plt.ylabel('salary')
plt.show()

"""#Visualization of test set 
plt.scatter(X_test, y_test, color='red')
plt.plot(X_test , regressor.predict(X_test),color='blue')
plt.title('Salary vs Experession(Test set)')
plt.xlabel('Experience')
plt.ylabel('salary')
plt.show()
"""

File: Linear_Regression/Simple_linear_regression/Simple_linear_regression.R

dataset = read.csv('Salary_Data.csv')


library(caTools)
set.seed(123)
split =sample.split(dataset$Salary , SplitRatio = 2/3)
training_set = subset(dataset , split==TRUE)
test_set= subset(dataset , split==FALSE)

#fiting simple regression to the trainin set 

regressor = lm(formula = Salary ~ YearsExperience,
               data =training_set)

#predicting training set data

y_pred = predict(regressor , newdata = test_set)

#visualization of linear simple regresion

#install.packages('ggplot2')
library(ggplot2)

ggplot() +
  geom_point(aes(x=training_set$YearsExperience ,y = training_set$Salary),
             colour='red') +
  geom_line(aes(x=training_set$YearsExperience , y= predict(regressor , newdata=training_set)),
            colour = 'blue') +
  ggtitle('Salary vs Experience(Training set)') +
  xlab('years of experience') +
  ylab('salary')


  